src/utils/transition-manager.js

```diff
@@ -0,0 +1,209 @@
+  }
+
+  // Process the vewiport change, either ignore or trigger a new transiton.
+  processViewportChange(nextProps) {
+    const isTransitionInProgress = this._isTransitionInProgress();
+    const isTransitionEnabled = this._isTransitionEnabled(nextProps);
+
+    // NOTE: Be cautious re-ordering statements in this function.
+    if (!isTransitionEnabled && !isTransitionInProgress ||
+      this._shouldIgnoreViewportChange(nextProps)) {
+      this.props = nextProps;
+      return;
+    }
+
+    if (isTransitionEnabled) {
+      const currentViewport = this.transitionContext.viewport || extractViewportFrom(this.props);
+      const endViewport = this.transitionContext.endViewport;
+
+      const startViewport = this._shouldSnapToEnd() ?
+        (endViewport || currentViewport) :
+        currentViewport;
+
+      this._triggerTransition(startViewport, nextProps);
+
+      if (isTransitionInProgress) {
+        this.props.onTransitionInterrupt();
+      }
+      nextProps.onTransitionStart();
+    } else if (isTransitionInProgress) {
+      this.props.onTransitionInterrupt();
+      this._endTransition();
+    }
+
+    this.props = nextProps;
+  }
+
+  // Helper methods
+
+  /* eslint-disable max-depth */
+  _areViewportsEqual(startViewport, endViewport) {
+    for (const p of VIEWPORT_PROPS) {
+      if (Array.isArray(startViewport[p])) {
+        for (let i = 0; i < startViewport[p].length; ++i) {
+          if (startViewport[p][i] !== endViewport[p][i]) {
+            return false;
+          }
+        }
+      } else if (startViewport[p] !== endViewport[p]) {
+        return false;
+      }
+    }
+    return true;
+  }
+  /* eslint-enable max-depth */
+
+  _createTransitionInterval(delay) {
+    if (this.transitionContext.interval) {
+      clearInterval(this.transitionContext.interval);
+    }
+    return setInterval(() => this._updateViewport(), delay);
+  }
+
+  _endTransition() {
+    clearInterval(this.transitionContext.interval);
+    this.transitionContext = {
+      time: 0,
+      delta: 0,
+      interval: null,
+      viewport: null,
+      startViewport: null,
+      endViewport: null
+    };
+  }
+
+  _isTransitionInProgress() {
+    return this.transitionContext.interval;
+  }
+
+  _isTransitionEnabled(props) {
+    return props.transitionDuration !== 0;
+  }
+
+  _isUpdateDueToCurrentTransition(nextProps) {
+    if (this.transitionContext.viewport) {
+      const newViewport = extractViewportFrom(nextProps);
+      return this._areViewportsEqual(newViewport, this.transitionContext.viewport);
+    }
+    return false;
+  }
+
+  _shouldIgnoreViewportChange(nextProps) {
+    // Ignore update if it is due to current active transition.
+    if (this._isUpdateDueToCurrentTransition(nextProps)) {
+      return true;
+    }
+
+    // Ignore update if it is requested to be ignored
+    if (this._isTransitionInProgress() &&
+      this.props.transitionInterruption === TRANSITION_EVENTS.IGNORE) {
```

Comment:
Nit: Why do we use this.props here instead of nextProps? Shouldn't the most recently supplied props determine the behavior?

Reply:
When current props has `IGNORE`, it will ignore any new viewport changes, until the current transition is complete.